### DISCUSSION TITLE
Why can't I use two mutually exclusive Query at the same time.

### DISCUSSION THREAD

**Discussion by droog126** [bevyengine/bevy#4805] [Q&A] [answered]
## What problem or need does this solve?
I have two instances of the same type, with children under each instance, I need to use
```rust
let mut parent_query=paramSet.p0();
let mut children_query=paramSet.p1();
for (children) in parent_query.iter_mut(){
  if  let OK (chilrenProps) in children_query.get_mut(children[0])
}
```
Why can't I use two mutually exclusive queries at the same time, this is unreasonable.
##What solution do you want?

To distinguish Entity such as **ParentEntity** **childrenEntity** type, we cannot guarantee that ordinary queries are mutually exclusive, but Parent and children may guarantee that these two are mutually exclusive queries.

##What alternatives have you considered?

Additional solutions to address and/or address the issues raised.

## additional context

any other information you would like to add, such as relevant prior work,
Screenshots, benchmarks, and more

**Comment by alice-i-cecile** [comment, marked as answer]
Can you show the type signature of your system? Parents are commonly children themselves, in a nested hierarchy, so there's no guarantee that we're accessing distinct entities here.

A more general solution likely involves https://github.com/bevyengine/bevy/issues/3742, but for now, I would probably recommend using `Query::get_many_mut`.

**Reply by droog126** [reply]
> 

```rust
 mut set: ParamSet<(
        Query<
            (
                &Children,
            ),
            With<SnakeTag>,
        >,
        Query<&mut Props, With<(Parent,SnakeChildrenTag)>>,
    )>,
```
almost so like this.

**Reply by droog126** [reply]
> Can you show the type signature of your system? Parents are commonly children themselves, in a nested hierarchy, so there's no guarantee that we're accessing distinct entities here.
> 
> A more general solution likely involves #3742, but for now, I would probably recommend using `Query::get_many_mut`.

I learn form here https://bevy-cheatbook.github.io/features/parent-child.html.
Can a system have multiple queries at the same time?

**Reply by droog126** [reply]
in a way that conflicts with a previous system parameter. Consider using `Without<T>` to create disjoint Queries or merging conflicting Queries into a `ParamSet`,I seem to know what to do.

**Reply by easynam** [reply]
A system can have multiple queries - in your case, you can’t use both queries mutably at the same time because they're in a ParamSet together.

Since the queries don’t access any conflicting data (one accesses &Children and the other accesses &mut Props), you should be able to use 2 separate Query parameters instead, which will allow both to be used at the same time.
